### PR TITLE
Format code cells as a list for better diff

### PR DESCRIPTION
**Pull request by vidma**
uses the multiline format from `.ipynb`

e.g. earlier `snb` format was:
```javascript
{
  "source" : "val sum = dataset.map(_._2).reduce(_+_)\n\nprintln(sum)\n"
}
```
changed to more diffable and readable multiline:
```javascript
{
  "source" : [
    "val sum = dataset.map(_._2).reduce(_+_)\n",
    "\n",
    "println(sum)\n"
  ],
}
```

### CODE REVIEW

File: notebooks/core/Simple Spark.snb.ipynb [hunk 1]
```diff
@@ -19,122 +19,154 @@
...
-      "trusted" : true,
-      "input_collapsed" : false,
-      "collapsed" : true,
-      "id" : "73F60C7C9F2945E38AC0C252F2C3AC1E"
+      "cell_type" : "code",
+      "source" : [
+        "val sum = dataset.map(_._2).reduce(_+_)\n",
+        "\n",
+        "println(sum)\n"
+      ],
```
Comment (vidma) [inline notebooks/core/Simple Spark.snb.ipynb]: above is how the multiline source looks now...

**Comment by vidma** [comment]
P.S. this makes new notebook unreadable by older versions of Spark Notebook (ipython/jupyther had this multiline source long ago).

so maybe we should at least add/bump `nbformat_minor` (?)...

**Comment by vidma** [comment]
actually, there's not much use in bumping the version, 
either case the old SNC won't be able to parse the new format with `["line1\n", "line2"]`. but that's the price we have to pay.

```
notebook.NotebookDeserializationError: Failed to parse JSON: {"obj.cells[2].source":[{"msg":"error.expected.jsstring","args":[]}],"obj.cells[3].source":[{"msg":"error.expected.jsstring","args":[]}],"obj.cells[4].source":[{"msg":"error.expected.jsstring","args":[]}]}
```

**Comment by drewnoff** [comment]
Maybe to add an option to convert to old format, but I'm not sure whether this has any value

**Comment by vidma** [comment]
> Maybe add an option to convert to old format, but I'm not sure whether this has any value.

Apparently, this is slightly more complicated. 
to make it configurable via `application.conf`, it would introduce a new dependency in `spark-notebook-core` sub-project on the play framework or would need many lines of code change.

so, for now, I simply added a constant in `NBSerializer.scala` which need to be modified and SN recompiled. IMHO, should be good for now.

**Comment by vidma** [comment]
should be good to 🚢 now...

File: modules/core/src/main/scala/notebook/NBSerializer.scala [hunk 2]
```diff
@@ -127,7 +127,10 @@ object NBSerializer extends Logging {
 
   // read source as a string "line1\nline2" or a ["line1\n", "line2"]
   val readSourceFromList = (JsPath \ "source").read[List[String]]
-  val readSourceFromString = (JsPath \ "source").read[String].map(_.split("\n").map(_ + "\n").toList)
+  val readSourceFromString = (JsPath \ "source").read[String]
+    // split by newline, and keep the separator, see https://stackoverflow.com/a/2206432/1276782
+    .map(_.split("(?<=\\n)").toList)
+
```
Comment (vidma) [inline modules/core/src/main/scala/notebook/NBSerializer.scala]: P.S. the earlier one, with each save/read cycle it was adding an unwanted extra newline at the end of codecell...

**Pull request merged**